astropy/io/fits/tests/test_connect.py

```diff
@@ -414,6 +414,42 @@ def test_mask_str_on_read(self, tmp_path):
         tab = Table.read(filename, mask_invalid=False)
         assert tab.mask is None
 
+    def test_write_object_tables_with_unified(self, tmp_path):
+        """
+        Write objects with the unified I/O interface.
+        See https://github.com/astropy/astropy/issues/1906
+        """
+        filename = tmp_path / "test_table_object.fits"
+        # Make a FITS table with an object column
+        tab = Table([{"col1": None}])
```

Comment:
Thanks @saimn for your [answer](https://github.com/astropy/astropy/pull/14578#discussion_r1160858396) and your patience... 
May I ask why here the `[...]` are needed?
If I run
```py
from astropy.table import Table
Table({'col1': None}).write('/tmp/tmp.fits', format='fits')
```
the following error is thrown
```pytr
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/cmarmo/software/astropy/astropy/table/table.py", line 870, in __init__
    init_func(data, names, dtype, n_cols, copy)
  File "/home/cmarmo/software/astropy/astropy/table/table.py", line 1417, in _init_from_dict
    self._init_from_list(data_list, names, dtype, n_cols, copy)
  File "/home/cmarmo/software/astropy/astropy/table/table.py", line 1239, in _init_from_list
    self._init_from_cols(cols)
  File "/home/cmarmo/software/astropy/astropy/table/table.py", line 1459, in _init_from_cols
    lengths = {len(col) for col in cols}
              ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/cmarmo/software/astropy/astropy/table/table.py", line 1459, in <setcomp>
    lengths = {len(col) for col in cols}
               ^^^^^^^^
TypeError: len() of unsized object
```
Is that expected? Should I modify this test too?

Reply:
Yeah I'm also confused by this one (but it was dinner time ;)). I actually don't know what `Table([{"col1": None}])` is supposed to do (or `Table([{"col1": some_array}])`). I may have to ping @taldcroft or @mhvk here :)
The docs (https://docs.astropy.org/en/latest/table/construct_table.html#construct-table) mention various ways to create a `Table`, with a list of columns, a dict of columns, and a list of dict but this case is only with `rows=`:
```
data = [{'a': 5, 'b': 10},
        {'a': 15, 'b': 20}]
t = Table(rows=data)
```
It seems that `Table` does the same thing if `rows=` is used or not:
```
In [11]: Table([{"col1": None, "col2": [1, 2]}])
Out[11]: 
<Table length=1>
 col1    col2  
object int64[2]
------ --------
  None   1 .. 2

In [12]: Table(rows=[{"col1": None, "col2": [1, 2]}])
Out[12]: 
<Table length=1>
 col1    col2  
object int64[2]
------ --------
  None   1 .. 2
```
So instead of `Table([{'col1': None}])` I would use `Table({'col1': [None]})` which seems more natural.

Reply:
Yes, that's how things work, and I agree with using `Table({'col1': [None]})`